tests/ui/linkage-attr/incompatible-flavor.rs

```diff
@@ -1,5 +1,5 @@
 //@ compile-flags: --target=x86_64-unknown-linux-gnu -C linker-flavor=msvc --crate-type=rlib
-//@ needs-llvm-components:
+//@ needs-llvm-components: x86
```

Comment:
incredibly weird that `//@ needs-llvm-components:` was accepted at all, I thought we linted on that

Reply:
There's also:
https://github.com/rust-lang/rust/blob/5795086bdfe7ed988aa53a110bd0692c33d8755b/src/tools/tidy/src/target_specific_tests.rs#L69-L72

Which doesn't check that the list of components is non-empty either